api/v1alpha3/azuremachinetemplate_conversion.go

```diff
@@ -44,6 +51,25 @@ func (src *AzureMachineTemplate) ConvertTo(dstRaw conversion.Hub) error { // nol
 		}
 	}
 
+	if restored.Spec.Template.Spec.Image.SharedGallery != nil {
+		if sharedGalleryPublisher, ok := src.Annotations[sharedGalleryPublisherAnnotation]; ok {
+			dst.Spec.Template.Spec.Image.SharedGallery.Publisher = &sharedGalleryPublisher
+			delete(dst.Annotations, sharedGalleryPublisherAnnotation)
+		}
+		if sharedGalleryOffer, ok := src.Annotations[sharedGalleryOfferAnnotation]; ok {
+			dst.Spec.Template.Spec.Image.SharedGallery.Offer = &sharedGalleryOffer
+			delete(dst.Annotations, sharedGalleryOfferAnnotation)
+		}
+		if sharedGallerySKU, ok := src.Annotations[sharedGallerySKUAnnotation]; ok {
+			dst.Spec.Template.Spec.Image.SharedGallery.SKU = &sharedGallerySKU
+			delete(dst.Annotations, sharedGallerySKUAnnotation)
+		}
+
```

Comment:
Following the comment below, this can be simplified to:
```
if restored.Spec.Template.Image != nil && restored.Spec.Template.Image.SharedGallery != nil {
dst.Spec.Template.Image.SharedGallery. = restored.Spec.Template.Image.SharedGallery 
}
```

Reply:
good call, this works well.